temporal-test-server/src/test/java/io/temporal/testserver/functional/NexusWorkflowTest.java

```diff
@@ -54,432 +54,1039 @@
 public class NexusWorkflowTest {
   @Rule
   public SDKTestWorkflowRule testWorkflowRule =
-      SDKTestWorkflowRule.newBuilder().setDoNotStart(true).build();
+      SDKTestWorkflowRule.newBuilder()
+          .setDoNotStart(true)
+          .setWorkflowTypes(EchoNexusHandlerWorkflowImpl.class)
+          .build();
+
+  private final Payload defaultInput =
+      Payload.newBuilder().setData(ByteString.copyFromUtf8("input")).build();
+  private Endpoint testEndpoint;
+  private final String testService = "test-service";
+  private final String testOperation = "test-operation";
 
   @Before
-  public void checkExternal() {
+  public void setup() {
     // TODO: remove this skip once 1.25.0 is officially released and
     // https://github.com/temporalio/sdk-java/issues/2165 is resolved
     assumeFalse(
         "Nexus APIs are not supported for server versions < 1.25.0",
         testWorkflowRule.isUseExternalService());
+
+    testEndpoint = createEndpoint("nexus-workflow-test-endpoint");
```

Comment:
Nit: make this a class constant

Do you clean up this endpoint?

Reply:
It depends on the namespace + task queue getting set up for the suite so I was getting an NPE when trying to make it a constant.

I don't clean it up but I didn't think that would matter for in-memory tests?

Reply:
Sorry I meant make `"nexus-workflow-test-endpoint"` a constant

Well when we run on the real server https://temporalio.atlassian.net/browse/SDK-2560 it will matter right, or is the plan to address clean up as part of https://temporalio.atlassian.net/browse/SDK-2560.